augur/tasks/util/collection_util.py

```diff
@@ -425,3 +433,155 @@ def start_block_of_repos(logger,session,condition,limit,phases,repos_type,hook="
+            cond_list.extend(additional_conditions)
+
+        condition = and_(*cond_list)
+
+        #Order by the relevant weight for the collection hook
+        order = getattr(CollectionStatus,f"{hook}_weight" )
+
+        #start repos for new primary collection hook
+        collection_size = start_block_of_repos(
+            session.logger, session,
+            condition,
+            limit, phase_list, repos_type="new", hook=hook,sort=order
+        )
+        
+        #Update limit with amount of repos started
+        limit -= collection_size
+
+    #Now start old repos if there is space to do so.
+
+    #Get a list of all users.
+    query = s.sql.text("""
+        SELECT  
+        user_id
+        FROM augur_operations.users
+    """)
+
+    user_list = session.execute_sql(query).fetchall()
+    random.shuffle(user_list)
+
+    #Extract the user id from the randomized list and split into four chunks
+    split_user_list = split_list_into_chunks([row[0] for row in user_list], 4)
+
+    for quarter_list in split_user_list:
+
+        #Break out if limit has been reached
+        if limit > 0:
+
+            #Query a set of valid repositories sorted by weight, also making sure that the repos aren't new or errored
+            repo_query = s.sql.text(f"""
+                SELECT
+	            repo.repo_id
+                FROM augur_operations.user_groups 
+                JOIN augur_operations.user_repos ON augur_operations.user_groups.group_id = augur_operations.user_repos.group_id
+                JOIN augur_data.repo ON augur_operations.user_repos.repo_id = augur_data.repo.repo_id
+                JOIN augur_operations.collection_status ON augur_operations.user_repos.repo_id = augur_operations.collection_status.repo_id
+                WHERE user_id IN :list_of_user_ids AND {desired_status}='Success'
+                ORDER BY {hook}_weight
+                LIMIT :limit_num
+            """).bindparams(list_of_user_ids=tuple(quarter_list),limit_num=(limit*2))
+
+            valid_repos = session.execute_sql(repo_query).fetchall()
+            valid_repo_git_ids = [repo[0] for repo in valid_repos]
+
+
+            #only start repos older than the specified amount of days
+            cutoff_date = datetime.datetime.now() - datetime.timedelta(days=days_until_collect_again)
+
+            #Create conditions for this query of valid repos to make sure they are old enough and valid
+            collected_before = getattr(CollectionStatus,f"{hook}_data_last_collected" ) != None
+            make_sure_valid_repo = CollectionStatus.repo_id.in_(list(valid_repo_git_ids))
+            old_enough = getattr(CollectionStatus,f"{hook}_data_last_collected" ) <= cutoff_date
+            cond_list = [not_erroed, not_collecting,collected_before,old_enough,make_sure_valid_repo]
+
+
+            #Order by the relevant weight for the collection hook
+            order = getattr(CollectionStatus,f"{hook}_weight" )
+            
```

Comment:
I think a `weight` would be more descriptive variable name, because `sort=order` doesn't make much sense to me